cli/src/utils.rs

```diff
@@ -98,3 +102,57 @@ pub fn read_secret(secret: bool, unsafe_secret: Option<String>) -> eyre::Result<
         unsafe_secret.unwrap()
     })
 }
+
+/// Find and parse out all the remappings for the projects
+pub fn find_remappings(
+    libs: &[PathBuf],
+    remappings: &[Remapping],
+    remappings_txt: &Path,
+    remappings_env: &Option<String>,
+) -> Vec<Remapping> {
+    /// Helper function for parsing newline-separated remappings
+    fn remappings_from_newline(remappings: &str) -> impl Iterator<Item = Remapping> + '_ {
+        remappings.split('\n').filter(|x| !x.is_empty()).map(|x| {
+            Remapping::from_str(x).unwrap_or_else(|_| panic!("could not parse remapping: {}", x))
+        })
+    }
+
+    let mut result: Vec<_> = libs.iter().flat_map(Remapping::find_many).collect();
+
+    result.extend_from_slice(remappings);
+
+    // extend them with the one via the env vars
+    if let Some(ref env) = remappings_env {
+        result.extend(remappings_from_newline(env))
+    }
+
+    // extend them with the one via the requirements.txt
+    if let Ok(ref remap) = std::fs::read_to_string(remappings_txt) {
+        result.extend(remappings_from_newline(remap))
+    }
+
+    // remove any potential duplicates
+    result.sort_unstable();
+    result.dedup();
+
+    result
+}
+
+/// Find libraries for the project
+pub fn find_libs(root: &Path, lib_paths: &[PathBuf], hardhat: bool) -> Vec<PathBuf> {
+    if lib_paths.is_empty() {
+        if hardhat {
+            return vec![root.join("node_modules")]
+        }
+
+        // no libs directories provided
+        return ProjectPathsConfig::find_libs(&root)
+    }
+
+    let mut libs = lib_paths.to_owned();
```

Comment:
I believe that's more idiomatic
```suggestion
    let mut libs = lib_paths.to_vec();
```